src/main/java/xiaobai/TaskList.java

```diff
@@ -0,0 +1,117 @@
+package xiaobai;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A list of tasks.
+ * Supports operations: add, remove, mark, unmark, and render.
+ */
+public class TaskList {
+    private final List<Task> tasks;
+
+    /**
+     * Creates an empty TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Creates a TaskList with an initial list of tasks.
+     * If the provided list is null, an empty list is used.
+     *
+     * @param initial Initial task list.
+     */
+    public TaskList(List<Task> initial) {
+        this.tasks = new ArrayList<>(initial == null ? List.of() : initial);
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return Number of tasks.
+     */
+    public int size() {
+        return tasks.size();
+    }
+
+    /**
+     * Returns the task at the given 1-based index.
+     *
+     * @param index1Based 1-based index of the task.
+     * @return Task at the given index.
+     */
+    public Task get(int index1Based) {
```

Comment:
Would it be better to identify what this method gets
```suggestion
    public Task getTask(int index1Based) {
```